File: 3543/3543.cs
public class Solution {
    public int MaximumWeightedKedgePath(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MaximumWeightedKedgePath(root.left), MaximumWeightedKedgePath(root.right));
    }
}

File: 3543/3543.go
func maximumWeightedKedgePath(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := maximumWeightedKedgePath(root.Left)
	r := maximumWeightedKedgePath(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}